src/Parser/DestructionWarlock/Modules/SoulShards/SoulShardEvents.js

```diff
@@ -0,0 +1,254 @@
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS';
+import HIT_TYPES from 'Parser/Core/HIT_TYPES';
+import Enemies from 'Parser/Core/Modules/Enemies';
+import RESOURCE_TYPES from 'common/RESOURCE_TYPES';
+
+const debug = false;
+
+const SUMMON_COST = 10;
+const CHAOS_BOLT_COST = 20;
+const RAIN_OF_FIRE_COST = 30;
+const MAX_FRAGMENTS = 50;
+const FRAGMENTS_PER_SHARD = 10;
+
+const FRAGMENT_GENERATING_ABILITIES = [
+  SPELLS.IMMOLATE_DEBUFF.id,
+  SPELLS.CONFLAGRATE.id,
+  SPELLS.INCINERATE.id,
+  //Dimensional Rift would also be here, but that is evaluated on cast, not damage and would proc a lot of times
+];
+
+const FRAGMENT_SPENDING_ABILITIES  = {
```

Comment:
This is also defined in SoulShardTracker, is there a reason it cannot be shared?

Reply:
That's true, I'll try to look at if something couldn't be factored out (but not sure about this, since I'm using those as object literals here, with fragments associated, whereas in Tracker, I only care about the ID, and I get the fragments from the event)